### PR TITLE
feat: get or create weaviate user for curated discovery

### PR DESCRIPTION
**Pull request by mc-jones**
This PR:

- Requires a user to be logged in to use the curated discovery app. 
- creates a `weaviate` from the `gravity` user, if one doesn't already exist
- Adds backend support for user related actions

https://github.com/artsy/force/assets/38149304/7dd6ed82-be82-4e7b-a946-ed68f1a72581

### CODE REVIEW

File: src/Apps/ArtAdvisor/07-Curated-Discovery/types.ts
Comment (mc-jones) [inline src/Apps/ArtAdvisor/07-Curated-Discovery/types.ts:1]: Wasn't sure if we wanted to stick with a `types` file or just include in a component and export. Not strongly opinionated on this.

File: src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx [hunk 1]
```diff
@@ -27,6 +29,37 @@ export const Form: React.FC<FormProps> = props => {
...
+        createWeaviateUser()
+      }
+
+      if (!response.ok) {
+        throw new Error(
+          `Error checking for weaviate user: ${response.statusText}`
+        )
+      }
+    })
+  }, [user])
```
Comment (mc-jones) [inline src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx:62]: Some slightly awkward promise chaining here. But seemed the simplest approach. Happy to change if there is a preferred alternative.

File: src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx [hunk 2]
```diff
@@ -62,6 +95,14 @@ export const Form: React.FC<FormProps> = props => {
 
   return (
     <Box>
+      {!user && (
+        <AuthDialog
+          onClose={() => {
+            /** noop */
+          }}
+        />
```
[the remark this answers is not in the document]
Comment (anandaroop) [reply src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx:103]: We can roll with this for now. A simpler alternative _might_ be to redirect to the full-page login form instead, i.e. do this before the render:

```ts
if (!user) {
  router.replace("/login?redirectTo=/advisor/7")
}
```

**Review by anandaroop — APPROVED** [review]
Looks good! Just some minor Qs below…

File: src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts [hunk 3]
```diff
@@ -1,18 +1,25 @@
...
 const DEFAULT_ARTICLES_CLASS = "DiscoveryArticles"
+const DEFAULT_USERS_CLASS = "DiscoveryUsers"
 
 export class WeaviateDB {
   private client: WeaviateClient
   private artworkClass: string
   private marketingCollectionClass: string
   private articlesClass: string
+  private artworksClass: string
+  private userClass: string
```
[the remark this answers is not in the document]
Comment (mc-jones) [reply src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts:22]: Yeah, this seems to have snuck in from before we synced yesterday. Will remove

File: src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mc-jones) [reply src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx:103]: Thats definitely better. Good call. Will update.

[the remark this answers is not in the document]
Comment (mc-jones) [reply src/Apps/ArtAdvisor/07-Curated-Discovery/Components/Form/Form.tsx]: updated in [b308ad1](https://github.com/artsy/force/pull/14083/commits/b308ad143277135dceab7496ea9940731135132d)

**Pull request merged**